### ISSUE TITLE
Allow Patching of Shipping Options

### ISSUE THREAD

**Issue by paymentplugins** [paypal/paypal-checkout-components#1341]
### 🚀 Feature Proposal

Upon review of the PayPal Order docs, it does not seem possible to update the shipping options of an order.

[PayPal Order API Docs](https://developer.paypal.com/docs/api/orders/v2/#orders_patch)

### Motivation

This feature is needed because when a customer selects a shipping address, it is common for the available shipping options to change. The available shipping options presented to the customer in the PayPal Checkout window should reflect the available options for the most recently selected shipping address.

### Example

`paypal.Buttons({
    onShippingChange: (data, actions) => {
        // shipping address is in different state, so provide updated shipping options
       if(data.shipping_address.state == 'CA'){
             // add new shipping option for California
             actions.order.patch([
                   {
                         "op": "add",
                         "path": "/purchase_units/@reference_id==\'default\'/shipping/options",
                         "value": {
                                 // shipping data
                          }
                   }
             ]);
       }
    }
}).render('#paypal-container);`

**Update**
https://github.com/paypal/paypal-checkout-components/blob/master/docs/implement-checkout.md#onshippingchange

https://github.com/paypal/paypal-checkout-components/issues/1308

**Update2**
I have had some success testing the following path: `/purchase_units/@reference_id==\'default\'/shipping/options`. 
It doesn't update the PayPal Checkout window on every update though.

**Comment by bluepnume** [comment]
@rajarampadmanathan I think this is possible already right with /v2/orders patch? Do you have an example you can share?

@amaclean2 can you check into the page updating issue mentioned by @paymentplugins?

**Comment by paymentplugins** [comment]
@bluepnume  Thanks for the reply. 

@amaclean2 mentioned a patch that was released last Friday 6/19/20 that appears to resolve the update issue.  [#1308](https://github.com/paypal/paypal-checkout-components/issues/1308)

I have replied on that thread regarding another issue with `add` operation. Basically, when the `onShippingChange` event is triggered the first time, the pop up is not displaying the initial list of shipping options that would be associated with the buyer's default address.

If it's easier I can close this feature request since #1308 is being tracked.
